packages/components/src/components/pin-code/pin-code-utils.ts

```diff
@@ -0,0 +1,48 @@
+import type { FormState } from '../../utils/form/form-state';
+import { removeAttribute, setAttribute } from '../../utils';
+
+export const PIN_CODE_TYPES = ['number', 'password'] as const;
+export type PinCodeType = (typeof PIN_CODE_TYPES)[number];
+
+export type PinCodeUpdateEvent = { value: string | number };
+
+export type PinCodeState = FormState;
+
+export const isTypeNumber = (type: string): boolean => {
+  return type === 'number';
+};
+
+export const inputIsSingleDigit = (input: string): boolean => input.length === 1 && /\d/.test(input);
+
+export const joinInputValues = (pinCodeElements: HTMLInputElement[]): string =>
+  pinCodeElements.map((el) => el.value).join('');
+
+export const initHiddenInput = (
+  host: HTMLElement,
+  value: string,
+  disabled: boolean,
+  required: boolean
+): HTMLInputElement => {
+  const hiddenInput = document.createElement('input');
+  setAttribute(hiddenInput, 'name', 'hidden-input');
+  setAttribute(hiddenInput, 'aria-hidden', 'true');
+  setAttribute(hiddenInput, 'slot', 'hidden-input');
+  setAttribute(hiddenInput, 'tabindex', '-1');
+  syncHiddenInput(hiddenInput, value, disabled, required);
+  host.prepend(hiddenInput);
+  return hiddenInput;
+};
+
+export const syncHiddenInput = (hiddenInput: HTMLInputElement, value, disabled: boolean, required: boolean): void => {
+  setAttribute(hiddenInput, 'value', `${value}`);
```

Comment:
Does this work and correctly update the value so that the from receives it?

Reply:
yes, it does :)